devdox/app/schemas/repo.py

```
@@ -83,59 +84,112 @@ class GitRepoResponse(BaseModel):
+            "visibility": extracted_visibility,
+            "created_at": project.created_at,
+            "star_count": project.star_count,
+            "http_url_to_repo": project.http_url_to_repo,
+            "statistics": extracted_statistics,
+        }
+
+    @classmethod
+    def from_gitlab(cls, data: Project | SimpleNamespace | dict) -> GitRepoResponse | None:
+        if not data:
+            return None
+        elif isinstance(data, Project) or isinstance(data, SimpleNamespace):
```

Ruff (0.11.9)

Multiple `isinstance` calls for `data`, merge into a single call

Merge `isinstance` calls for `data`

(SIM101)